### PR TITLE
ja: translate Ch. 41 and 42

### PR DESCRIPTION
**Pull request by HidenoriKobayashi**
Japanese translation for Ch. 41 & 42.

Part of #652.

### CODE REVIEW

**Comment by HidenoriKobayashi** [comment]
Sorry, I had mistakenly put both ch. 41 & 42 into one commit. If that makes it hard to review, please let me know.

**Comment by keiichiw** [comment]
@kantasv would you mind helping review? I'm going to take a look later anyway, though

**Comment by kantasv** [comment]
Sure thing, @keiichiw -san. I'll take a look into this :)

File: po/ja.po [hunk 1]
```diff
@@ -12606,42 +12706,56 @@ msgid ""
...
+"LSM303AGRドライバは`embedded_hal::blocking::i2c::WriteRead`を実装するものを必"
+"要とします。 [`microbit::hal::Twim`](https://docs.rs/microbit-v2/latest/"
+"microbit/hal/struct.Twim.html)構造体がこれを実装しています。"
 
 #: src/exercises/bare-metal/compass.md:17
 msgid ""
 "You have a [`microbit::Board`](https://docs.rs/microbit-v2/latest/microbit/"
 "struct.Board.html) struct with fields for the various pins and peripherals."
 msgstr ""
+"様々なピンや周辺のための [`microbit::Board`](https://docs.rs/microbit-v2/"
```
Comment (kantasv) [inline po/ja.po:12739]: [nit] "周辺I/O" (as consistently used in your translation) or "ペリフェラル" instead of "周辺"?

File: po/ja.po [hunk 2]
```diff
@@ -12606,42 +12706,56 @@ msgid ""
...
 msgid ""
 "The LSM303AGR Inertial Measurement Unit is connected to the internal I2C bus."
-msgstr ""
+msgstr "LSM303AGR慣性計測器は内部のI2Cバスに接続されています。"
 
 #: src/exercises/bare-metal/compass.md:12
 msgid ""
 "TWI is another name for I2C, so the I2C master peripheral is called TWIM."
-msgstr ""
+msgstr "TWIはI2Cの別名なので、I2Cマスタ周辺はTWIMという名前になっています。"
```
Comment (kantasv) [inline po/ja.po:12722]: Is the translation "peripheral" -> "周辺" common in the microcontroller community?

File: po/ja.po [hunk 3]
```diff
@@ -12110,13 +12119,15 @@ msgid ""
...
 #: src/bare-metal/microcontrollers/mmio.md:66
 #: src/bare-metal/microcontrollers/pacs.md:59
 #: src/bare-metal/microcontrollers/hals.md:43
 #: src/bare-metal/microcontrollers/board-support.md:34
 msgid "Run the example with:"
-msgstr ""
+msgstr "例の実行方法:"
 
 #: src/bare-metal/microcontrollers/pacs.md:1
 msgid "Peripheral Access Crates"
```
Comment (kantasv) [inline po/ja.po:12133]: [Suggestion] adding translation such as "周辺I/Oへアクセスするためのクレート（PACs）"?

File: po/ja.po [hunk 4]
```diff
@@ -12413,26 +12470,33 @@ msgid ""
 "[probe-rs](https://probe.rs/) is a handy toolset for embedded debugging, "
 "like OpenOCD but better integrated."
 msgstr ""
+"[probe-rs](https://probe.rs/)は組み込み開発に有用なツールセットです。これは"
```
Comment (kantasv) [inline po/ja.po:12473]: IMO, "組み込み向けデバッグ" if we want to keep the nuances in the original sentence and the fact that`probe-rs` is for embedded debugging? (Of course, *development* itself entails debugging as a part of it.)

File: po/ja.po [hunk 5]
```diff
@@ -12413,26 +12470,33 @@ msgid ""
...
 
 #: src/bare-metal/microcontrollers/probe-rs.md:10
 msgid ""
 "`cargo-embed` is a cargo subcommand to build and flash binaries, log RTT "
 "(Real Time Transfers) output and connect GDB. It's configured by an `Embed."
 "toml` file in your project directory."
 msgstr ""
+"`cargo-embed`はcargoのサブコマンドであり、バイナリをビルドしたり、フラッシュ"
+"したり、RTT（Real Time Transfers）の出力ログを取得したり、GDBに接続するための"
+"ものです。設定はあなたのプロジェクトディレクトリにおける`Embed.toml`ファイル"
```
Comment (kantasv) [inline po/ja.po:12498]: Optional: may drop "あなた" in JA translation?

File: po/ja.po [hunk 6]
```diff
@@ -12484,44 +12562,48 @@ msgstr ""
...
+"を提供"
 
 #: src/bare-metal/microcontrollers/other-projects.md:6
 msgid "[Embassy](https://embassy.dev/)"
 msgstr ""
 
 #: src/bare-metal/microcontrollers/other-projects.md:7
 msgid "`async` executors with priorities, timers, networking, USB"
-msgstr ""
+msgstr "優先度付き`async` 実行、タイマ、ネットワーク、USB"
```
Comment (kantasv) [inline po/ja.po:12606]: "優先度付き`async` 実行" -> "優先度付き`async`エグゼキュータ" for consistency?

[FYI] "Executor" -> "エグゼキュータ" translations found:
- In our course: [here](https://google.github.io/comprehensive-rust/ja/async/async-await.html#:~:text=%E3%81%9F%E3%82%81%E3%81%AB%E3%81%AF%E3%80%81-,%E3%82%A8%E3%82%B0%E3%82%BC%E3%82%AD%E3%83%A5%E3%83%BC%E3%82%BF,-%E3%81%8C%E5%BF%85%E8%A6%81%E3%81%A7%E3%81%99)
- In Rustbook: [here](https://async-book-ja.netlify.app/02_execution/04_executor#:~:text=Programming%20in%20Rust-,%E3%82%A8%E3%82%B0%E3%82%BC%E3%82%AD%E3%83%A5%E3%83%BC%E3%82%BF%E3%83%BC,-%E3%82%92%E6%9B%B8%E3%81%84)

**Review by kantasv — COMMENTED** [review]
So far, I have went thru half of the MR diff, and sharing some thoughts. Most of my comments are optional suggestions regarding consistency but not too technical this time.

**Comment by kantasv** [comment]
Hi, @HidenoriKobayashi -san. I went thru all the diff. Kindly have a look my comments :) I handle a lot of hardware components on a daily basis, so the section was very interesting.

File: po/ja.po (hunk 2 above)
Comment (HidenoriKobayashi) [reply po/ja.po:12722]: Hmm, I grew up with that Japanese term. Although, for this specific sentence, I could probably omit it.

File: po/ja.po (hunk 4 above)
Comment (HidenoriKobayashi) [reply po/ja.po:12473]: Sounds good. Thanks, I'll take your suggestion!

File: po/ja.po (hunk 5 above)
Comment (HidenoriKobayashi) [reply po/ja.po:12498]: Maybe. I left it to be clear that you need to edit the one in your local tree, but I also had a feeling that was unnatural as Japanese. Let me think of something that sounds more natural.

File: po/ja.po (hunk 6 above)
Comment (HidenoriKobayashi) [reply po/ja.po:12606]: Thanks! I didn't know about such translations.

File: po/ja.po (hunk 3 above)
Comment (HidenoriKobayashi) [reply po/ja.po:12133]: Sure.

**Comment by HidenoriKobayashi** [comment]
Thanks @kantasv -san! I have taken your suggestions with a minor tweak. Please resolve the conversations if you are happy with the latest version.

File: po/ja.po [hunk 7]
```diff
@@ -12017,28 +12018,36 @@ msgid ""
 "The `cortex_m_rt` crate provides (among other things) a reset handler for "
 "Cortex M microcontrollers."
 msgstr ""
+"クレートの `cortex_m_rt`はCortex Mマイクロコントローラ向けのリセットハンドラ"
```
Comment (keiichiw) [inline po/ja.po:12021]:
```suggestion
" `cortex_m_rt`クレートはCortex Mマイクロコントローラ向けのリセットハンドラ"
```

File: po/ja.po [hunk 8]
```diff
@@ -12017,28 +12018,36 @@ msgid ""
...
 msgid "Run the example with `cargo embed --bin minimal`"
-msgstr ""
+msgstr "この例は`cargo embed --bin minimal`により実行します"
 
 #: src/bare-metal/microcontrollers/mmio.md:3
 msgid ""
 "Most microcontrollers access peripherals via memory-mapped IO. Let's try "
 "turning on an LED on our micro:bit:"
 msgstr ""
+"大半のマイクロコントローラはメモリマップされたIO空間を通して周辺I/Oにアクセス"
```
Comment (keiichiw) [inline po/ja.po:12049]:
```suggestion
"大半のマイクロコントローラはメモリマップドIOを通して周辺I/Oにアクセス"
```
I think「メモリマップドIO」is a common translation of memory-mapped IO
https://ja.wikipedia.org/wiki/%E3%83%A1%E3%83%A2%E3%83%AA%E3%83%9E%E3%83%83%E3%83%97%E3%83%89I/O

File: po/ja.po [hunk 9]
```diff
@@ -12110,24 +12119,29 @@ msgid ""
...
 
 #: src/bare-metal/microcontrollers/pacs.md:3
 msgid ""
 "[`svd2rust`](https://crates.io/crates/svd2rust) generates mostly-safe Rust "
 "wrappers for memory-mapped peripherals from [CMSIS-SVD](https://www.keil.com/"
 "pack/doc/CMSIS/SVD/html/index.html) files."
 msgstr ""
+"[`svd2rust`](https://crates.io/crates/svd2rust) は[CMSIS-SVD](https://www."
+"keil.com/pack/doc/CMSIS/SVD/html/index.html) ファイルから、メモリマップされた"
+"周辺I/Oに対するほぼ安全（safe）なRustラッパーを生成します。"
```
Comment (keiichiw) [inline po/ja.po:12144]:
```suggestion
"周辺I/Oに対するほぼ安全（mostly-safe）なRustラッパーを生成します。"
```

File: po/ja.po [hunk 10]
```diff
@@ -12250,42 +12277,52 @@ msgstr ""
...
 msgstr ""
+"この例では、ボードサポートクレートは単に分かりやすい名前を提供し、少しの初期"
+"化を実施しているだけです。"
 
 #: src/bare-metal/microcontrollers/board-support.md:30
 msgid ""
 "The crate may also include drivers for some on-board devices outside of the "
 "microcontroller itself."
 msgstr ""
+"マイクロコントローラの外に実装されたボードレベル・デバイスに対するドライバも"
```
Comment (keiichiw) [inline po/ja.po:12316]: Question:  Is it common to translate "on-board devices" to "ボードレベル・デバイス"? I thought it's "オンボードデバイス" but I'm not so familiar with this area, so I'm not confident

File: po/ja.po [hunk 11]
```diff
@@ -12413,26 +12470,33 @@ msgid ""
...
 
 #: src/bare-metal/microcontrollers/probe-rs.md:7
 msgid "GDB stub and Microsoft DAP (Debug Adapter Protocol) server"
-msgstr ""
+msgstr "GDBスタブやMicrosoft DAP (Debug Adapter Protocol)サーバ"
 
 #: src/bare-metal/microcontrollers/probe-rs.md:8
 msgid "Cargo integration"
-msgstr ""
+msgstr "Cargo統合"
```
Comment (keiichiw) [inline po/ja.po:12488]: I'd say "Cargoとのインテグレーション" or "Cargo との統合"

**Review by keiichiw — APPROVED** [review]
Thank you so much! I had only a few comments

**Comment by kantasv** [comment]
Hi, @HidenoriKobayashi -san, thanks for having a look at my review and fixing the po file as in [e7fe0a0](https://github.com/google/comprehensive-rust/pull/1442/commits/e7fe0a0f2807dabfaa63713e685721085da56a32). 

Looks like I can't resolve the conversation --- only those who submitted the MR can resolve conversation? But overall, I agree with your chnages in e7fe0a0. Feel free to assume that all my suggestions were resolved by the commit :)
>Please resolve the conversations if you are happy with the latest version.

File: po/ja.po (hunk 10 above)
Comment (chikoski) [reply po/ja.po:12316]: +1 to オンボードデバイス. According to [IT media](https://atmarkit.itmedia.co.jp/icd/root/25/5785825.html), オンボードデバイス is the translation of onboard device, which indicates devices being embedded in a board.

File: po/ja.po [hunk 12]
```diff
@@ -12250,42 +12277,52 @@ msgstr ""
...
 "The crate may also include drivers for some on-board devices outside of the "
 "microcontroller itself."
 msgstr ""
+"マイクロコントローラの外に実装されたボードレベル・デバイスに対するドライバも"
+"提供されていることがあります。"
 
 #: src/bare-metal/microcontrollers/board-support.md:32
 msgid "`microbit-v2` includes a simple driver for the LED matrix."
-msgstr ""
+msgstr "`microbit-v2`はLEDマトリックスに対する簡単なドライバを含んでいます。"
```
Comment (chikoski) [inline po/ja.po:12321]: Perhaps, "マトリクスLED" could sound more natural to the audience than "LEDマトリクス". c.f. [Search result on Akizukidenshi](https://akizukidenshi.com/catalog/goods/search.aspx?search=x&keyword=%83%7D%83g%83%8A%83N%83X&image=%8C%9F%8D%F5)

**Review by chikoski — COMMENTED** [review]
Added some minor comments.

**Comment by HidenoriKobayashi** [comment]
Thanks everyone for the feedback!
If everything looks okay now, please merge!

**Comment by keiichiw** [comment]
Thanks! LGTM!

**Pull request merged**